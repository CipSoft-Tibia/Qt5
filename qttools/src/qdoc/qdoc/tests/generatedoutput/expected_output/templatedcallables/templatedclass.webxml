<?xml version="1.0" encoding="UTF-8"?>
<WebXML>
    <document>
        <class name="TemplatedClass" href="templatedclass.html" status="active" access="public" location="templated_callables.h" documented="true" module="TestQDoc">
            <description>
                <para>Containing record for the tested methods.</para>
            </description>
            <function name="templated_method_with_defaulted_non_type_template_parameter" fullname="TemplatedClass::templated_method_with_defaulted_non_type_template_parameter" href="templatedclass.html#templated_method_with_defaulted_non_type_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_defaulted_non_type_template_parameter()">
                <description>
                    <para>A templated method under a templated class with a defaulted non type template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_defaulted_template_template_parameter" fullname="TemplatedClass::templated_method_with_defaulted_template_template_parameter" href="templatedclass.html#templated_method_with_defaulted_template_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_defaulted_template_template_parameter(Container&lt;U, size&gt;)">
                <parameter type="Container&lt;U, size&gt;" name="" default=""/>
                <description>
                    <para>A templated method under a templated class with a defaulted template template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_defaulted_type_template_parameter" fullname="TemplatedClass::templated_method_with_defaulted_type_template_parameter" href="templatedclass.html#templated_method_with_defaulted_type_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_defaulted_type_template_parameter(U)">
                <parameter type="U" name="" default=""/>
                <description>
                    <para>A templated method under a templated class with a defaulted type template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_non_type_template_parameter" fullname="TemplatedClass::templated_method_with_non_type_template_parameter" href="templatedclass.html#templated_method_with_non_type_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_non_type_template_parameter()">
                <description>
                    <para>A templated method under a templated class with a non-defaulted non type template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_non_type_template_parameter_pack" fullname="TemplatedClass::templated_method_with_non_type_template_parameter_pack" href="templatedclass.html#templated_method_with_non_type_template_parameter_pack" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_non_type_template_parameter_pack()">
                <description>
                    <para>A templated method under a templated class with a non type template parameter pack.</para>
                </description>
            </function>
            <function name="templated_method_with_placeholder_non_type_template_parameter" fullname="TemplatedClass::templated_method_with_placeholder_non_type_template_parameter" href="templatedclass.html#templated_method_with_placeholder_non_type_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_placeholder_non_type_template_parameter()">
                <description>
                    <para>A templated method under a templated class with a placeholder non type template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_template_template_parameter" fullname="TemplatedClass::templated_method_with_template_template_parameter" href="templatedclass.html#templated_method_with_template_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_template_template_parameter(X&lt;U&gt;)">
                <parameter type="X&lt;U&gt;" name="" default=""/>
                <description>
                    <para>A templated method under a templated class with a non-defaulted template template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_template_template_parameter_pack" fullname="TemplatedClass::templated_method_with_template_template_parameter_pack" href="templatedclass.html#templated_method_with_template_template_parameter_pack" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_template_template_parameter_pack(Container&lt;U&gt;...)">
                <parameter type="Container&lt;U&gt;..." name="" default=""/>
                <description>
                    <para>A templated method under a templated class with a template template parameter pack.</para>
                </description>
            </function>
            <function name="templated_method_with_type_template_parameter" fullname="TemplatedClass::templated_method_with_type_template_parameter" href="templatedclass.html#templated_method_with_type_template_parameter" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_type_template_parameter(U)">
                <parameter type="U" name="" default=""/>
                <description>
                    <para>A templated method under a templated class with a non-defaulted type template parameter.</para>
                </description>
            </function>
            <function name="templated_method_with_type_template_parameter_pack" fullname="TemplatedClass::templated_method_with_type_template_parameter_pack" href="templatedclass.html#templated_method_with_type_template_parameter_pack" status="active" access="public" location="templated_callables.h" documented="true" meta="plain" type="void" signature="void templated_method_with_type_template_parameter_pack(Ts...)">
                <parameter type="Ts..." name="" default=""/>
                <description>
                    <para>A templated method under a templated class with a type template parameter pack.</para>
                </description>
            </function>
        </class>
    </document>
</WebXML>
