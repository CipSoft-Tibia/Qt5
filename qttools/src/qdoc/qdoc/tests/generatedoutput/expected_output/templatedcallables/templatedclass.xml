<?xml version="1.0" encoding="UTF-8"?>
<db:article xmlns:db="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.2" xml:lang="en">
<db:info>
<db:title>TemplatedClass Class</db:title>
<db:subtitle>template &lt;typename T&gt; class TemplatedClass</db:subtitle>
<db:productname>templatedcallables</db:productname>
<db:titleabbrev>templatedcallables Reference Documentation</db:titleabbrev>
<db:abstract>
<db:para>templatedcallables Reference Documentation.</db:para></db:abstract>
</db:info>
<db:variablelist>
<db:varlistentry>
<db:term>Header</db:term>
<db:listitem>
<db:para>TemplatedClass</db:para>
</db:listitem>
</db:varlistentry>
</db:variablelist>
<db:section xml:id="details">
<db:title>Detailed Description</db:title>
<db:para>Containing record for the tested methods.</db:para>
</db:section>
<db:section xml:id="member-function-documentation">
<db:title>Member Function Documentation</db:title>
<db:section xml:id="templated_method_with_defaulted_non_type_template_parameter">
<db:title>void TemplatedClass::templated_method_with_defaulted_non_type_template_parameter()</db:title>
<db:para>A templated method under a templated class with a defaulted non type template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_defaulted_template_template_parameter">
<db:title>void TemplatedClass::templated_method_with_defaulted_template_template_parameter(<db:emphasis>Container&lt;U, size&gt;</db:emphasis>)</db:title>
<db:para>A templated method under a templated class with a defaulted template template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_defaulted_type_template_parameter">
<db:title>void TemplatedClass::templated_method_with_defaulted_type_template_parameter(<db:emphasis>U</db:emphasis>)</db:title>
<db:para>A templated method under a templated class with a defaulted type template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_non_type_template_parameter">
<db:title>void TemplatedClass::templated_method_with_non_type_template_parameter()</db:title>
<db:para>A templated method under a templated class with a non-defaulted non type template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_non_type_template_parameter_pack">
<db:title>void TemplatedClass::templated_method_with_non_type_template_parameter_pack()</db:title>
<db:para>A templated method under a templated class with a non type template parameter pack.</db:para>
</db:section>
<db:section xml:id="templated_method_with_placeholder_non_type_template_parameter">
<db:title>void TemplatedClass::templated_method_with_placeholder_non_type_template_parameter()</db:title>
<db:para>A templated method under a templated class with a placeholder non type template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_template_template_parameter">
<db:title>void TemplatedClass::templated_method_with_template_template_parameter(<db:emphasis>X&lt;U&gt;</db:emphasis>)</db:title>
<db:para>A templated method under a templated class with a non-defaulted template template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_template_template_parameter_pack">
<db:title>void TemplatedClass::templated_method_with_template_template_parameter_pack(<db:emphasis>Container&lt;U&gt;...</db:emphasis>)</db:title>
<db:para>A templated method under a templated class with a template template parameter pack.</db:para>
</db:section>
<db:section xml:id="templated_method_with_type_template_parameter">
<db:title>void TemplatedClass::templated_method_with_type_template_parameter(<db:emphasis>U</db:emphasis>)</db:title>
<db:para>A templated method under a templated class with a non-defaulted type template parameter.</db:para>
</db:section>
<db:section xml:id="templated_method_with_type_template_parameter_pack">
<db:title>void TemplatedClass::templated_method_with_type_template_parameter_pack(<db:emphasis>Ts...</db:emphasis>)</db:title>
<db:para>A templated method under a templated class with a type template parameter pack.</db:para>
</db:section>
</db:section>
</db:article>
